### PR TITLE
feat(ci_visibility): add support for AWS codepipeline to CI env var gathering

### PR DESCRIPTION
**Pull request by romainkomorndatadog**
This adds support for gathering CI environment variables for AWS Codepipeline (although it's noteworthy that most of the environment variables need to be provided by the user).

## Checklist

- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed. If no release note is required, add label `changelog/no-changelog`.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).
- [x] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer has explicitly acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment.
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)
- [x] If this PR touches code that signs or publishes builds or packages, or handles credentials of any kind, I've requested a review from `@DataDog/security-design-and-guidance`.
- [x] This PR doesn't touch any of that.

### CODE REVIEW

File: ddtrace/ext/ci.py [hunk 1]
```diff
@@ -542,4 +559,5 @@ def extract_buddy(env):
     ("TRAVIS", extract_travis),
     ("BITRISE_BUILD_SLUG", extract_bitrise),
     ("BUDDY", extract_buddy),
+    ("CODEBUILD_INITIATOR", extract_awscodepipeline),
```
Comment (anmarchenko) [inline ddtrace/ext/ci.py:562]: I think the spec says that we should not only check for CODEBUILD_INITIATOR env variable but also that it starts with `codepipeline `, so I have to refactor the similar logic in the ruby libarry

Comment (romainkomorndatadog) [reply ddtrace/ext/ci.py:562]: Oh man, 100% missed that. Thank you! Will update.

Comment (romainkomorndatadog) [reply ddtrace/ext/ci.py:562]: Okay, it's updated. I was originally thinking a refactor would be nice (kind of like you did in https://github.com/DataDog/datadog-ci-rb/pull/52 , @anmarchenko ) but in the end, the behavior is the same: the tags aren't added, so I figured leaving things as is would be simpler.

Comment (romainkomorndatadog) [reply ddtrace/ext/ci.py:562]: My `git push` failed and I didn't pay attention. 👎 , sorry. It's coming.

**Review by anmarchenko — APPROVED** [review]

**Review by juan-fernandez — APPROVED** [review]

**Review by majorgreys — APPROVED** [review]

**Pull request merged**